### ISSUE TITLE
feature request: parseable output

### ISSUE THREAD

**Issue by barryw** [1N3/Sn1per#66]
This is a great tool, but the output is really hard to parse. I understand that it's running a collection of disparate tools, and each has its own output, but what would be nice is a uniform output in something structured (xml, json, etc) so that we could use it as part of an automated process.

Is that even remotely possible?

**Comment by GitHipster** [comment]
I am struggling with the same thing. Structured xml output would be great indeed.
Until there is a structured output mechanism in sniper (if it even ends up as a feature) i can offer you my approach. Using Excel and leveraging formula's i run sniper and collect the results almost automatically. The following lines run sniper, save the output to the desktop, convert the terminal output to HTML and clean up the leftovers in the sniper directory:

```
mkdir /root/Desktop/sniper/www.github.com
cp -r /usr/share/sniper/loot/* /root/Desktop/sniper/www.github.com
mkdir /root/Desktop/sniper/www.github.com/_backup
cd /root/Desktop/sniper/www.github.com/
cp *.txt /root/Desktop/sniper/www.github.com/_backup/
for i in ?????????????????*.txt; do mv -i "$i" "${i%?????????????????}"; done
mv /root/Desktop/sniper/www.github.com/sniper-www.github.com /root/Desktop/sniper/www.github.com/sniper-www.github.com.txt
cat /root/Desktop/sniper/www.github.com/sniper-www.github.com.txt | aha --black > sniper-www.github.com.html

rm /usr/share/sniper/loot/domains/*
rm /usr/share/sniper/loot/nmap/*
rm /usr/share/sniper/loot/output/*
rm /usr/share/sniper/loot/reports/*
rm /usr/share/sniper/loot/screenshots/*
rm /usr/share/sniper/loot/*.txt
rm /usr/share/sniper/loot/notes/*

nautilus /root/Desktop/sniper/

firefox /root/Desktop/sniper/www.github.com/sniper-www.github.com.html
```
The only thing is that output isn't converted to HTML properly keeping the color coding. The sniper 'logo' is in red when running sniper. It's uncolored in the HTML file. Most of the findings / interesting results are colored correctly.

Maybe you can put these couple of lines to good use. 
You will need to install aha of course. This can be done by:

`apt-get install aha`

**Comment by 1N3** [comment]
Hey guys, thanks for the feedback and suggestions... This is definitely on the road map to work on so I'll keep this open until that work is done.

**Comment by amanelis** [comment]
+1

**Comment by 1N3** [comment]
Hi, Sn1per Professional and Sn1per Community Edition 5.0 have been released which save the output of several tools to the workspace directory (/usr/share/sniper/loot/). Sn1per Professional, which is available from the xerosecurity.com website (https://xerosecurity.com) takes this a step further by providing a GUI interface with the scan results to easily parse through each host. This should help finding the data you want more easily. Let me know if you have any questions.
